sql/catalyst/src/test/scala/org/apache/spark/sql/RowTest.scala

```diff
@@ -108,6 +108,31 @@ class RowTest extends AnyFunSpec with Matchers {
     }
   }
 
+  describe("row not equals") {
+    val externalRow = Row(1)
+    val externalRow2 = Row(1.0)
+
+    it("inequality check for external rows") {
+      externalRow should not equal externalRow2
+    }
```

Comment:
Hmm, isn't this somehow arguable? Especially 1 == 1.0 => true.

Reply:
Yes, I was thinking about this, Seq(1) == Seq(1.0) is also true.